### ISSUE TITLE
mrconvert -export_grad_* and b-value scaling

### ISSUE THREAD

**Issue by bjeurissen** [MRtrix3/mrtrix3#980]
I noticed today that in the current master branch ``mrinfo -export_grad_fsl`` performs b-value scaling, while ``mrconvert dwi.mif dwi.nii -export_grad_fsl`` does not.

As a consequence, in cases where the b-values have to be scaled by the gradient magnitude, ``dwipreproc`` [will fall over at the eddy stage](https://github.com/MRtrix3/mrtrix3/blob/master/bin/dwipreproc#L458) (eddy throws ``KMatrixException: msg=NewSphericalKMatrix::GetHyperParGuess:  Unable to find valid hyperparameters``).

This issue is related to #178 

In order to fix this, we should either:

- make ``mrconvert`` treat the ``-export_grad_fsl`` option in the exact same way as ``mrinfo``. This includes adding a ``-raw_dwgrad`` to override b-value scaling.

- change the ``dwipreproc`` script to use ``mrinfo`` instead of ``mrconvert`` to extract the gradient directions.

**Comment by jdtournier** [comment]
OK, I think I'd agree with your first option. The conversion to bvecs/bvals is primarily there to support interoperation with other packages, so we should make sure we provide the information in a manner consistent with our own interpretation. I suppose we could make the same argument for `-export_grad_mrtrix`, but I'm not sure about that one - the interoperability argument doesn't apply in the same way there - I think I'd still export that version as per the input. 

One slight issue here is that the way `mrconvert` handles this currently is that it passes the information through with no checking or modification of any sort. That includes not checking whether the number of entries matches the number of volumes, normalisation of the vectors, and b-value scaling. If we do decide to add b-value scaling for `-export_grad_fsl`, that will most likely also drag in these other checks. Probably no bad thing given that the bvecs/bvals are specific for the corresponding image - but it is a minor difference.

**Comment by bjeurissen** [comment]
I am really unsure how to tackle this. If we would do this for ``-export_grad_fsl``, I think we have to do this for ``-export_grad_mrtrix`` as well, otherwise we are not being consistent. One could even argue that we would have to do it then also when exporting to .mif, just to remain consistent...

**Comment by Lestropie** [comment]
I don't agree that exporting to FSL-style or MRtrix-style files needs to be consistent; they're two different formats with different requirements. Our only obligation is to make sure that the output of `-export_grad_fsl` plays ball with FSL.

If the bvecs / bvals format explicitly requires normalised vectors, this should probably be being applied deep, i.e. within the `save_bvecs_bvals()` function itself, so it's impossible to bypass when writing to that format.
